### ISSUE TITLE
Branch strategy for sustainable feature development

### ISSUE THREAD

**Issue by yizha1** [notaryproject/.github#49]
Currently Notation v1.0.0 release is blocked by [branding issue](https://github.com/notaryproject/.github/issues/35). While fixing the naming issues and it takes some time, the feature development for v1.1.0 release should continue. The proposal is to create a release branch `release-1.0` based off the main branch per [release management](https://github.com/notaryproject/notation/blob/main/RELEASE_MANAGEMENT.md#patch-releases), so that v1.1.0 feature development can be unblocked. This branch `release-1.0` is also used for patch releases (bug fixes for 1.0.0 features or security issues) in later phases. We should also document this strategy for sustainable feature development. What I am not sure is how to handle the [version file](https://github.com/notaryproject/notation/blob/main/internal/version/version.go) and the `1.0.0` tag of `main` branch

/cc @priteshbandi @iamsamirzon @shizhMSFT @toddysm @FeynmanZhou @sajayantony for comments.

**Comment by yizha1** [comment]
What we discussed is to address the following scenario:

All the code and documentation for a release are merged into the main branch. It is ready to cut a release from main. However, the release is blocked for other issues not related to the release scope, for example, branding issue blocks the Notation v1 release.

The solution is to create a release branch, so that new features can be developed continuously in the main branch.
The detailed how-to is addressed by https://github.com/notaryproject/notation/pull/713
